client/src/root.js

```diff
@@ -35,7 +45,117 @@ if (mailtrainConfig.reportsEnabled) {
+            state.setIn(['namespace', 'error'], t("namespaceHasToBeSelectedToApply"));
+        } else {
+            state.setIn(['namespace', 'error'], null);
+        }
+    }
+
+    componentDidMount() {
+        var enabled = false;
+        if(getNamespaceIdFilterCookie()){
+            enabled = true;
+        }
+        this.populateFormValues({
+            namespace: getNamespaceIdFilterCookie(),
+            namespace_filter_enabled: enabled
+        });
+        this.loadTreeData();
+    }
+
+    @withAsyncErrorHandler
+    async loadTreeData() {
+
+            const response = await axios.get(getUrl('rest/namespaces-tree'));
+            const data = response.data;
+            for (const root of data) {
+                root.expanded = true;
+            }
+
+            if (this.props.entity && !this.isEditGlobal()) {
+                this.removeNsIdSubtree(data);
+            }
+
+            if (this.isComponentMounted()) {
+                this.setState({
+                    treeData: data
+                });
+            }
+        
+    }
+
+    async setNamespaceFilterAsync() {
+        const t = this.props.t;
+
+        if(this.getFormValue('namespace_filter_enabled')){
+            if(this.getFormValue('namespace')){
+                document.cookie = "namespaceFilterId=" + this.getFormValue('namespace') + "; path=/";
+                document.cookie = "namespaceFilterName=" + this.getFormValue('namespace') + "; path=/";
+                this.state.namespaceId = this.getFormValue('namespace');
+                this.state.namespaceName = this.getFormValue('namespace');
+                this.setFormStatusMessage('warning', null);
+                this.props.onHide();
+            }else{
+                this.setFormStatusMessage('warning', t('namespaceMustNotBeEmpty'));
+            }
+        }else{
+            deleteNamespaceFilterCookies();
+            this.state.namespaceId = null;
+            this.state.namespaceName = "Namespace filter";
+            this.setFormStatusMessage('warning', null);
+            this.props.onHide();
+        }
+    }
+
+    async hideModal() {
+        this.setFormStatusMessage('warning', null);
+        this.props.onHide();
+    }
+
+    render() {
+        const t = this.props.t;
+
+        return (
+            <ModalDialog hidden={!this.props.visible} title={t('namespaceFilter')} onCloseAsync={() => this.hideModal()} buttons={[
+                { label: t('apply'), className: 'btn-primary', onClickAsync: ::this.setNamespaceFilterAsync },
+                { label: t('close'), className: 'btn-danger', onClickAsync: ::this.hideModal }
+            ]}>
+                <Form stateOwner={this}>
+                    <TreeTableSelect id="namespace" format="wide" label={t('namespace')} data={this.state.treeData}/>
+                    <hr/>
+                    <CheckBox id="namespace_filter_enabled" format="wide" text={t('enabled')}></CheckBox>
```

Comment:
Personally, I would put the checkbox above it and I would show the TreeTableSelect only when the checkbox is checked.

Reply:
Ok, that's a good point